### ISSUE TITLE
Automated file upload

### ISSUE THREAD

**Issue by bradjones452** [ignacionelson/ProjectSend#490]
I was wondering if there is a way to do automated uploads from an application. Pass the account info and the file info and then upload the file to the ProjectSend site. We want to use the same product to do GUI uploads as well as an application(automated) upload. 

Can anyone point me in the right direction?

Thanks Brad Jones.

**Comment by onlinestatements** [comment]
+1 for this request

**Comment by kkplein** [comment]
Would be useful, yes.

**Comment by onlinestatements** [comment]
Post your reply on the website under my request. If there is enough people to say they want it then they would be more willing to help maybe.On Nov 10, 2017 9:41 AM, kkplein <notifications@github.com> wrote:Would be useful, yes.

**Comment by ignacionelson** [comment]
Hello! It's not possible right now. I don't think it would be too hard to implement, but it would heavily depend on your server configuration and available modules. I'm doing the last touches on the upcoming version, after that I'll try to give this a go!

**Comment by onlinestatements** [comment]
I think i have cent os with cpanel. With php 7.xI am using php 5 right now because php 7 makes the server run out of ram which makes my website go down.ThanksOn Nov 14, 2017 7:18 PM, Ignacio Nelson <notifications@github.com> wrote:Hello! It's not possible right now. I don't think it would be too hard to implement, but it would heavily depend on your server configuration and available modules. I'm doing the last touches on the upcoming version, after that I'll try to give this a go!

**Comment by bradjones452** [comment]
Also I thought if you allow a url to be used for the auto upload

ex. http://projectsend.somedomain.com/upload?username=xxx?password=yyyy?filename=xxx.txt
and assign the file to the username passed.
Just a thought... Brad Jones.

**Comment by indprodaja** [comment]
+1

**Comment by glmdev** [comment]
This, please. +1

**Comment by emmenlau** [comment]
This would be a very relevant feature to integrate ProjectSend with i.e. build/continuous integration systems.

**Comment by onlinestatements** [comment]
We are a printing and mailing company.
I just had a software developer that owns a billing software company and has many mutual clients with our company and is asking if there is a way to do what this thread is asking?
He is redesigning the software and making it cloud based and wants to know if there is a way to upload files to our website without the customer having to download the data file then manually upload it to our project send website.

Do you know if this will be possible to do soon?

Thank You

**Comment by Blazefoxar** [comment]
no solution yet?

**Comment by vjirinec** [comment]
Any update on this?  We are very interested in automating this.

**Comment by glmdev** [comment]
Unfortunately, this project appears to have been untouched for nearly a year now.

**Comment by glego** [comment]
Hi Guys,

I've created a hacky way to upload the files from bash. Take care to install curl and jq.

https://gist.github.com/glego/3cdc74e14b79673d261ecccf1e3bfdcc

Regards,
Glenn

**Comment by emmenlau** [comment]
Cool! This is actually quite nice. I like that it does not require any changes to ProjectSend. Which version of ProjectSend are you using?

**Comment by glego** [comment]
The latest r1070 version

**Comment by schnudd31do3** [comment]
With r1295 it is not working.

There is still no file transfer possible at file-level. Because the filenames are hashed you can not automate it at file level. Best would be to keep original filenames in the upload folder. The file information (description, owner, upload-date, group membership etc.) could be kept and supplied in a separate folder and info-file, which is automatically managed also by projectsend. The .htaccess and index.php also should not be in the upload/files folder in order to by able to synchronize with network shares.

**Comment by MassimoMessore** [comment]
This would be a very useful feature also for continuous delivery scenarios. BTW in the case this feature will be eventually integrated, I suggest to avoid the direct use of username and password but use personal access tokens managed by administrators instead.

**Comment by iincitr** [comment]
Yes, it will be very useful for our solutions. Our implementation via FTP
transfers every 10 min to upload/files folder and then assign user/client
file(s) manually.

**Comment by onlinestatements** [comment]
Does this hacky way work with version 1605?

> Hi Guys,
> 
> I've created a hacky way to upload the files from bash. Take care to install curl and jq.
> 
> https://gist.github.com/glego/3cdc74e14b79673d261ecccf1e3bfdcc
> 
> Regards, Glenn

**Comment by leknoppix** [comment]
> Hi Guys,
> 
> I've created a hacky way to upload the files from bash. Take care to install curl and jq.
> 
> https://gist.github.com/glego/3cdc74e14b79673d261ecccf1e3bfdcc
> 
> Regards, Glenn

"Thank you for this script; but is it still relevant?"

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
